src/test/java/io/cdap/plugin/ariba/source/PropertiesTest.java

```diff
@@ -0,0 +1,332 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.ariba.source;
+
+import com.google.gson.Gson;
+import io.cdap.cdap.api.data.schema.Schema;
+import io.cdap.cdap.etl.api.batch.BatchSource;
+import io.cdap.cdap.etl.api.connector.ConnectorContext;
+import io.cdap.cdap.etl.api.connector.ConnectorSpec;
+import io.cdap.cdap.etl.api.connector.ConnectorSpecRequest;
+import io.cdap.cdap.etl.api.connector.PluginSpec;
+import io.cdap.cdap.etl.api.validation.ValidationException;
+import io.cdap.cdap.etl.mock.common.MockConnectorConfigurer;
+import io.cdap.cdap.etl.mock.common.MockConnectorContext;
+import io.cdap.cdap.etl.mock.validation.MockFailureCollector;
+import io.cdap.plugin.ariba.source.AribaBatchSource;
+import io.cdap.plugin.ariba.source.AribaServices;
+import io.cdap.plugin.ariba.source.config.AribaPluginConfig;
+import io.cdap.plugin.ariba.source.connector.AribaConnector;
+import io.cdap.plugin.ariba.source.connector.AribaConnectorConfig;
+import io.cdap.plugin.ariba.source.exception.AribaException;
+import io.cdap.plugin.ariba.source.metadata.AribaColumnMetadata;
+import io.cdap.plugin.ariba.source.metadata.AribaResponseContainer;
+import io.cdap.plugin.ariba.source.metadata.proto.AribaMetaResponse;
+import io.cdap.plugin.ariba.source.metadata.proto.ArrayFields;
+import io.cdap.plugin.ariba.source.metadata.proto.ObjectFields;
+import io.cdap.plugin.ariba.source.metadata.proto.Properties;
+import io.cdap.plugin.ariba.source.metadata.proto.SimpleFields;
+import io.cdap.plugin.ariba.source.util.ResourceConstants;
+import io.cdap.plugin.common.ConfigUtil;
+import mockit.Expectations;
+import mockit.Mocked;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+public class PropertiesTest {

```

Comment:
Why was this test moved from metadata/proto/ to here?

Reply:
As some of the extra test cases has been added in this file regarding the connection management.

Reply:
I don't see any reason why the test needs to be moved to src/test/java/io/cdap/plugin/ariba/source.
Keep the test in src/test/java/io/cdap/plugin/ariba/source/metadata/proto/PropertiesTest.java and add new testcases there.